lib/practice_exercises.rb

```diff
@@ -1,13 +1,41 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) The while loop
+# Space Complexity: O(1), it doesn't require use of extra memory.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  valid_chars = string.gsub(/[^a-z^0-9]/i, '').downcase.split("")
+  i = 0
+  j = valid_chars.length - 1
+  while i < j
+    unless valid_chars[i] == valid_chars[j]
+      return false
+    end
+    i += 1
+    j -= 1
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n) Only one loop and depends on the size of the strings
+# Space Complexity: O(1) it doesn't require use of extra memory
 def longest_prefix(strings)
```

Comment:
The outer loop depends on the length of the 1st string, and the inner the number of strings, so this is O(nm) time complexity.

This method builds a prefix string so it has space complexity of O(m)